syntax = "proto3";
import "common_proto/tip.proto";
import "common_proto/common.proto";
import "component_proto/scene_comp.proto";
option cc_generic_services = true;

message EnterSceneC2SRequest
{
  SceneInfo scene_info = 1;
}

message EnterSceneC2SResponse
{
	Tips error = 1;
}

message EnterSceneS2C
{
  SceneInfo scene_info = 1;
}

message SceneInfoS2C
{
  repeated SceneInfo scene_info = 1;
}

service ClientPlayerSceneService 
{
  rpc EnterSceneC2S (EnterSceneC2SRequest) returns (EnterSceneC2SResponse);
  rpc PushEnterSceneS2C (EnterSceneS2C) returns (EnterSceneS2C);
  rpc PushSceneInfoS2C (SceneInfoS2C) returns (SceneInfoS2C);
}